src/game/objects/bridge.c

```diff
@@ -101,65 +113,105 @@ void DrawBridgeCollision(
 }
 
 void BridgeFlatFloor(
-    ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
+    ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int32_t *height)
 {
-    if (y <= item->pos.y) {
-        *height = item->pos.y;
+    if (g_Config.fix_bridge_collision && !IsSameBridgeSector(x, y, z, item)) {
+        return;
     }
+    if (y > item->pos.y) {
+        return;
+    }
+
+    *height = item->pos.y;
 }
 
 void BridgeFlatCeiling(
-    ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
+    ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int32_t *height)
 {
+    if (g_Config.fix_bridge_collision && !IsSameBridgeSector(x, y, z, item)) {
+        return;
+    }
+
     if (y > item->pos.y) {
         *height = item->pos.y + STEP_L;
     }
 }
 
-int32_t GetOffset(ITEM_INFO *item, int32_t x, int32_t z)
+int32_t GetOffset(ITEM_INFO *item, int32_t x, int32_t y, int32_t z)
 {
-    if (item->pos.y_rot == 0) {
-        return (WALL_L - x) & (WALL_L - 1);
-    } else if (item->pos.y_rot == -PHD_180) {
-        return x & (WALL_L - 1);
-    } else if (item->pos.y_rot == PHD_90) {
-        return z & (WALL_L - 1);
-    } else {
-        return (WALL_L - z) & (WALL_L - 1);
-    }
+    int32_t offset = 0;
+    if (item->pos.y_rot == 0)
```

Comment:
We **do not** omit curly braces. Never.

Reply:
Not intentional. I'll fix. I was testing Tomb5's version of this function.

Reply:
Fixed.